instrumentation/elasticsearch/elasticsearch-rest-6.4/javaagent/src/test/java/ElasticsearchRest6Test.java

```diff
@@ -0,0 +1,187 @@
+            .setMaxRetryTimeoutMillis(Integer.MAX_VALUE)
+            .setRequestConfigCallback(
+                builder ->
+                    builder
+                        .setConnectTimeout(Integer.MAX_VALUE)
+                        .setSocketTimeout(Integer.MAX_VALUE))
+            .build();
+
+    objectMapper = new ObjectMapper();
+  }
+
+  @AfterAll
+  static void cleanupSpec() {
+    elasticsearch.stop();
+  }
+
+  @Test
+  @SuppressWarnings({"deprecation", "rawtypes"})
+  // ignore deprecation interface
+  public void elasticsearchStatus() throws IOException {
+
+    Response response = client.performRequest("GET", "_cluster/health");
+    Map result = objectMapper.readValue(response.getEntity().getContent(), Map.class);
+
+    Assertions.assertEquals(result.get("status"), "green");
+
+    testing.waitAndAssertTraces(
+        trace -> {
+          trace.hasSpansSatisfyingExactly(
+              span -> {
+                span.hasName("GET")
+                    .hasKind(SpanKind.CLIENT)
+                    .hasNoParent()
+                    .hasAttributesSatisfyingExactly(
+                        equalTo(SemanticAttributes.DB_SYSTEM, "elasticsearch"),
+                        equalTo(SemanticAttributes.DB_OPERATION, "GET"),
+                        equalTo(SemanticAttributes.DB_STATEMENT, "GET _cluster/health"));
+              },
+              span -> {
+                span.hasName("GET")
+                    .hasKind(SpanKind.CLIENT)
+                    .hasParent(trace.getSpan(0))
+                    .hasAttributesSatisfyingExactly(
+                        equalTo(SemanticAttributes.NET_PEER_NAME, httpHost.getHostName()),
+                        equalTo(SemanticAttributes.NET_PEER_PORT, httpHost.getPort()),
+                        equalTo(SemanticAttributes.HTTP_METHOD, "GET"),
+                        equalTo(AttributeKey.stringKey("net.protocol.name"), "http"),
+                        equalTo(AttributeKey.stringKey("net.protocol.version"), "1.1"),
+                        equalTo(SemanticAttributes.HTTP_URL, httpHost.toURI() + "/_cluster/health"),
+                        equalTo(SemanticAttributes.HTTP_STATUS_CODE, 200L),
+                        equalTo(
+                            SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH,
+                            response.getEntity().getContentLength()));
+              });
+        });
+  }
+
+  @Test
+  @SuppressWarnings({"deprecation", "rawtypes"})
+  // ignore deprecation interface
+  public void elasticsearchStatusAsync() throws Exception {
+    Response[] requestResponse = {null};
+    Exception[] exception = {null};
+    CountDownLatch countDownLatch = new CountDownLatch(1);
+    ResponseListener responseListener =
+        new ResponseListener() {
+          @Override
+          public void onSuccess(Response response) {
+            testing.runWithSpan(
+                "callback",
+                () -> {
+                  requestResponse[0] = response;
```

Comment:
We create the `callback` span here to verify that context is propagated into the `onSuccess` method. You can move the `requestResponse[0] = response;` out of the `runWithSpan`, then you don't need to do the array trick and can use `requestResponse = response;`.

Reply:
Even if I move it to the `onSuccess` method, it still seems to need to use the array.

Reply:
sorry my bad